### ISSUE TITLE
Super class that extends from Presented*****Layout

### ISSUE THREAD

**Issue by jjvargas** [lukaspili/Mortar-architect#17]
There is no multiple inheritance, but I would like to know if somehow it's possible to create a base class that extends from any of the Presented****Layouts.

**Comment by lukaspili** [comment]
I'm not sure I understand. Can you write a sample code, even if it's not compiler-valid?

**Comment by jjvargas** [comment]
So I have two different Views

```
public class SearchGridMusicView extends PresentedLinearLayout
public class HomeView extends PresentedScrollView
```

And I want to create a base class for both 

```
public class BaseView extends PresentedLinearLayout,PresentedScrollView
```

So I can use it like

```
public class SearchGridMusicView  extends BaseView 
public class HomeView extends BaseView 
```

**Comment by lukaspili** [comment]
Yeah this is the problem with inheritance (and Android relies much on it), because it's not flexible at all.
The only easy fix I see is for your 2 views to have the same base layout, like a `FrameLayout`. And then each one adds its custom layout `LinearLayout` and `ScrollView`.
The drawback would be that Android will render a useless FrameLayout, but the cost is negligible.

I don't know what's in your `BaseView` class, but it may be possible to apply the **composition pattern** rather than inheritance. Also, if you are a Kotlin user, you would may be able to use Traits to solve this :)
